src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -326,7 +388,97 @@ public void RequestLicKey(int dwReserved, [MarshalAs(UnmanagedType.BStr)] out st
                 [MarshalAs(UnmanagedType.BStr)] string bstrKey,
                 [MarshalAs(UnmanagedType.Interface)] out object ppvObject)
             {
-                throw new NotImplementedException();
+                Debug.Assert(pUnkReserved == null);
+                this.CreateInstanceInner(pUnkOuter, ref riid, bstrKey, isDesignTime: false, out ppvObject);
+            }
+
+            private void CreateInstanceInner(
+                object pUnkOuter,
+                ref Guid riid,
+                string key,
+                bool isDesignTime,
+                out object ppvObject)
+            {
+                BasicClassFactory.ValidateInterfaceRequest(this.classType, ref riid, pUnkOuter);
+
+                ppvObject = this.licenseProxy.AllocateAndValidateLicense(this.classType, key, isDesignTime);
+                if (pUnkOuter != null)
+                {
+                    ppvObject = BasicClassFactory.CreateAggregatedObject(pUnkOuter, ppvObject);
+                }
+            }
+        }
+
+        private class LicenseInteropProxy
+        {
+            private static readonly Type s_LicenseAttrType;
+            private readonly Type helperType;
+            private readonly MethodInfo getLicInfo;
+            private readonly MethodInfo requestLicKey;
+            private readonly MethodInfo allocateAndValidateLicense;
+            private readonly object instance;
+
+            static LicenseInteropProxy()
+            {
+                s_LicenseAttrType = Type.GetType("System.ComponentModel.LicenseProviderAttribute, System", throwOnError: true);
+            }
+
+            public LicenseInteropProxy()
+            {
+                Type licman = Type.GetType("System.ComponentModel.LicenseManager, System", throwOnError: true);
+                this.helperType = licman.GetNestedType("LicenseInteropHelper", BindingFlags.NonPublic);
+                this.getLicInfo = this.helperType.GetMethod("GetLicInfo2", BindingFlags.Instance | BindingFlags.NonPublic);
+                this.requestLicKey = this.helperType.GetMethod("RequestLicKey2", BindingFlags.Static | BindingFlags.NonPublic);
+                this.allocateAndValidateLicense = this.helperType.GetMethod("AllocateAndValidateLicense2", BindingFlags.Static | BindingFlags.NonPublic);
+
+                this.instance = Activator.CreateInstance(this.helperType);
+            }
+
+            public static bool HasLicense(Type type)
+            {
+                return type.IsDefined(s_LicenseAttrType, inherit: true);
+            }
+
+            public void GetLicInfo(Type type, out bool runtimeKeyAvail, out bool licVerified)
+            {
+                var parameters = new object[] { type, null, null };
+                try
+                {
+                    this.getLicInfo.Invoke(this.instance, parameters);
+                }
+                catch (TargetInvocationException e)
```

Comment:
You can pass `BindingFlags.DoNotWrapExceptions` to Invoke instead of doing this manual unwrapping.